### DISCUSSION TITLE
PyNaCl

### DISCUSSION THREAD

**Discussion by k98kurz** [flet-dev/flet#4316] [Packages] [answered]
PyPI link: https://pypi.org/project/PyNaCl/

From the PyPI description: "PyNaCl is a Python binding to [libsodium](https://github.com/jedisct1/libsodium), which is a fork of the [Networking and Cryptography library](https://nacl.cr.yp.to/). These libraries have a stated goal of improving usability, security and speed. It supports Python 3.6+ as well as PyPy 3."

Result of `pipgrip PyNaCl`:

```
pynacl==1.5.0
cffi==1.17.1
pycparser==2.22
```

The wheel files are platform specific.

I know that the cryptography package has been added, but it does not have the granular level of bindings necessary for a number of advanced cryptography uses, e.g. MuSig, adapter signatures, and homomorphic one-way cryptography constructions, all of which I have already implemented and published using PyNaCl. Since this may be a bit more of a niche package, I am willing to contribute to adding this package to Flet if pointed in the right direction.

**Comment by FeodorFitsner** [comment, marked as answer]
PyNaCl is available for iOS and Android
